iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/UnsecureConnectionCheck.java

```diff
@@ -38,15 +38,15 @@ public class UnsecureConnectionCheck implements IacCheck {
   private static final Set<String> SENSITIVE_CURL_OPTION = Set.of("-k", "--insecure", "--proxy-insecure", "--doh-insecure");
 
   private static final CommandDetector SENSITIVE_CURL_COMMAND = CommandDetector.builder()
-    .with("curl"::equals)
-    .withOptionalRepeating(s -> !SENSITIVE_CURL_OPTION.contains(s) && s.startsWith("-"))
-    .with(SENSITIVE_CURL_OPTION::contains)
+    .with("curl")
+    .withOptionalRepeating(s -> s.startsWith("-") && !SENSITIVE_CURL_OPTION.contains(s))
```

Comment:
As you use this concept multiple times maybe you can find a new method like `anyFlagFollowedBy`
which combines `.withOptionalRepeating(s -> s.startsWith("-") && !followingFlags.contains(s))` and `.with(followingFlags)`. Because you will only provide `followingFlags`. WDYT?